src/components/WalletAssets.vue

```diff
@@ -207,8 +204,44 @@ export default class WalletAssets extends Mixins(LoadingMixin, FormattedAmountMi
     }
   }
 
-  mounted(): void {
-    this.assetList = api.assets.accountAssets;
+  filterAssets(filters: WalletAssetFilters): void {
+    if (filters.option === WalletFilteringOptions.ALL) {
+      this.assetList = api.assets.accountAssets.filter((asset) => this.applyFilterOptions(asset, filters));
+    }
+
+    if (filters.option === WalletFilteringOptions.TOKEN) {
+      const tokens = api.assets.accountAssets.filter((asset) => !api.assets.isNft(asset));
```

Comment:
this code can be reused
```
const filteredTokens = (nft = false) => api.assets.accountAssets.filter((asset) => api.assets.isNft(asset) === nft)
...
const tokens = filteredTokens();
...
const nuts = filteredTokens(true);
```

Reply:
As for me it leads to poorer readability
Prefer not using flags and struggle to avoid them
Anyway, it can't be reused as it enters only one if-clause and then return